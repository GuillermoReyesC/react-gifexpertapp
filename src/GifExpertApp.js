import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ( { defaultCategories = [] } ) => {

    

    
    const [ categories, setCategories ] = useState( defaultCategories );

    //const [ categories, setCategories ] = useState(['One Punch']);
    
/*    // const categories = ['x', 'x2', 'x3', 'z']

  

   const handleAdd = () => {

   //setCategories(['n',...categories]);    ---> operador spread  " ...categories "
   setCategories( cats => [...categories, 'n'] );   // ---> arrow function para agregar a una const
   

   }
   console.log(categories) */

    return (
        <>
            <h2> GifExpertApp </h2>
            <AddCategory setCategories={ setCategories } />
            <hr />
            

           
            
                { 
                categories.map( category => (
                    <GifGrid 
                    key = { category }
                    category = { category }
                    />
                ))
                }
            

        </>
    )
}

export default GifExpertApp
