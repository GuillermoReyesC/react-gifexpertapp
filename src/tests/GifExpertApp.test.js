import { shallow } from 'enzyme';
import React from "react";
import {GifExpertApp} from '../GifExpertApp';


describe('pruebas al componente <GifExpertApp/>', () => {
    
    test('debe mostrarse correctamente <GifExpertApp',()=>{

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    }); 

    test('debe mostrarse correctamente <GifExpertApp con categorias',()=>{

        const categories = ['One Punch', 'Samurai X'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    }); 

 

   

        

});