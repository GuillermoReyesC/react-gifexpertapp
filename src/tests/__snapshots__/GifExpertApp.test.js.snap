// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas al componente <GifExpertApp/> debe mostrarse correctamente <GifExpertApp 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas al componente <GifExpertApp/> debe mostrarse correctamente <GifExpertApp con categorias 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
  </ol>
</Fragment>
`;
